lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/authentication/KerberosAuthentication.java

```diff
@@ -42,24 +42,18 @@ public KerberosAuthentication(KerberosConfiguration kerberosConfiguration)
         this.configuration = kerberosConfiguration.getConfiguration();
     }
 
-    public Subject getSubject()
+    public LoginContext getLoginContext()
     {
         Subject subject = new Subject(false, ImmutableSet.of(principal), emptySet(), emptySet());
-        try {
-            LoginContext loginContext = new LoginContext("", subject, null, configuration);
-            loginContext.login();
-            return loginContext.getSubject();
-        }
-        catch (LoginException e) {
-            throw new RuntimeException(e);
-        }
+        return getLoginContextFromSubject(subject);
     }
 
-    public void attemptLogin(Subject subject)
+    public LoginContext getLoginContextFromSubject(Subject subject)
     {
         try {
             LoginContext loginContext = new LoginContext("", subject, null, configuration);
             loginContext.login();
```

Comment:
nit: Since we are trying to do an login - should we rename the method ?

Reply:
Renamed.